testing/e2e/agent_container_test.go

```diff
@@ -186,5 +208,365 @@ func (suite *AgentContainerSuite) TestWithSecretFiles() {
+	})
+}
+
+// TestSkeep10m checks if the fleet-server is healthy after 10m
+// skipped unless the "-long" flag is provided.
+func (suite *AgentContainerSuite) TestSleep10m() {
+	if !longFlag {
+		suite.T().Skip("Long tests skipped. Enable with -long.")
+	}
+	bCtx, bCancel := context.WithCancel(context.Background())
+	defer bCancel()
+
+	req := testcontainers.ContainerRequest{
+		Hostname: "fleet-server",
+		Image:    suite.dockerImg,
+		Env: map[string]string{
+			"GOCOVERDIR":                       "/cover",
+			"FLEET_SERVER_ENABLE":              "1",
+			"FLEET_URL":                        "https://fleet-server:8220",
+			"FLEET_CA":                         "/tmp/e2e-test-ca.crt",
+			"FLEET_SERVER_CERT":                "/tmp/fleet-server.crt",
+			"FLEET_SERVER_CERT_KEY":            "/tmp/fleet-server.key",
+			"FLEET_SERVER_CERT_KEY_PASSPHRASE": "/tmp/passphrase",
+			"FLEET_SERVER_SERVICE_TOKEN":       suite.serviceToken,
+			"FLEET_SERVER_ELASTICSEARCH_HOST":  "http://elasticsearch:9200",
+			"FLEET_SERVER_POLICY_ID":           "fleet-server-policy",
+		},
+		ExposedPorts: []string{"8220/tcp"},
+		Networks:     []string{"integration_default"},
+		// certs are copied so they can be readable by fleet-server.
+		Files: []testcontainers.ContainerFile{{
+			HostFilePath:      filepath.Join(suite.certPath, "e2e-test-ca.crt"),
+			ContainerFilePath: "/tmp/e2e-test-ca.crt",
+			FileMode:          0644,
+		}, {
+			HostFilePath:      filepath.Join(suite.certPath, "fleet-server.crt"),
+			ContainerFilePath: "/tmp/fleet-server.crt",
+			FileMode:          0644,
+		}, {
+			HostFilePath:      filepath.Join(suite.certPath, "fleet-server.key"),
+			ContainerFilePath: "/tmp/fleet-server.key",
+			FileMode:          0644,
+		}, {
+			HostFilePath:      filepath.Join(suite.certPath, "passphrase"),
+			ContainerFilePath: "/tmp/passphrase",
+			FileMode:          0644,
+		}},
+		Mounts: testcontainers.ContainerMounts{
+			testcontainers.ContainerMount{
+				Source: &testcontainers.GenericBindMountSource{suite.coverPath},
+				Target: "/cover",
+			},
+		},
+	}
+	fleetC, err := testcontainers.GenericContainer(bCtx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+		Logger:           &logger{suite.T()},
+	})
+	suite.Require().NoError(err)
+	suite.container = fleetC
+
+	endpoint, err := fleetC.Endpoint(bCtx, "https")
+	suite.Require().NoError(err)
+
+	suite.FleetIsHealthy(bCtx, endpoint)
+
+	suite.T().Log("sleeping for 10m")
+	time.Sleep(time.Minute * 10)
+
+	suite.FleetIsHealthy(bCtx, endpoint)
+}
+
+// TestDockerAgent will enroll a real agent running in a docker container.
+// It uses the dummy-policy (no monitoring or integrations).
+// The test is successful if it reaches the "online" state in Kibana within 5 minutes of the container starting.
+//
+// NOTE: If we wanted to do more tests against an actual agent (such as sending actions or even running integrations) we should make it a test suite.
```

Comment:
I'm not sure if we want to test elastic-agent and fleet-server action handling (using the binaries) as part of this test suite or the one in the agent repo 
cc @cmacknz @pierrehilbert

Reply:
My opinion is that the server should test that it implements its own API contract correctly independent of any client implementation. The client implementation (Elastic Agent) is responsible for ensuring it interacts correctly with the server implementation (Fleet Server).

This would lead you to testing action handling and the other APIs in the Fleet Server repository, but using a perfect reference client that you wrote independently and not Elastic Agent itself.

The agent would then have an integration test against a real Fleet Server and Fleet Server doesn't depend on Elastic Agent itself for testing.

Reply:
I'll keep this test as a sanity check.
We'll use the generated API client for endpoint testing